File: Problem Set 1/Mario/mario.c
#include <cs50.h>
#include <stdio.h>

// Asks user for a number between 1 and 8
int main(void)
{
    int n;
    do
    {
        n = get_int("Choose a height between 1 and 8: ");
    }
    while (n < 1 || n > 8);
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            // determines how many spaces to print before the #s.
            if (j == n - i - 1 || j > n - i - 1)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        // prints the next loop on a new line.
        printf("\n");
    }
}

File: Problem Set 6/Mario/mario.py
from cs50 import get_int

def main():
    height = get_height()
    print_pyramid(height)


def get_height():
    while True:
        n = get_int("Enter a number between 1 and 8\n")
        if (n >= 1 and n <= 8):
            break
    return n

def print_pyramid(n):

    for i in range(n):
        print(" " * (n - (i + 1)), end="")
        print("#" * (i + 1), end="")
        print("  ", end="")
        print("#" * (i + 1), end="\n")

main()